### ISSUE TITLE
Object detection metrics are always -1

### ISSUE THREAD

**Issue by typical-byte-world** [tensorflow/models#7423]
I'm using Google Colab

I'm trying to train object detection model. I tried first ssdlite_mobilenet_v2_coco, after 40000 epochs the model did not predict any boxes. The output was the original picture. I found in the issues here on the github that this model does not work for many and I changed it to ssd_inception_v2_coco. After training it, I ran into a problem, the model predicts about 20 boxes throughout the image. During training, my metrics are always -1.
![Screenshot from 2019-08-09 10-38-51](https://user-images.githubusercontent.com/32809014/62777863-66eb7800-bab7-11e9-9fc8-45611548b4a2.png)
![Screenshot from 2019-08-09 10-38-41](https://user-images.githubusercontent.com/32809014/62777864-67840e80-bab7-11e9-97ee-481208ea64a6.png)
![Screenshot from 2019-08-09 10-16-19](https://user-images.githubusercontent.com/32809014/62777865-67840e80-bab7-11e9-841b-c887e8edb10c.png)

I tried to change the learning rate to 0.1, but it didn't work.

 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = -1.000

My labels:
item {
  id: 1
  name: 'Scoreboard'
  display_name: 'Scoreboard'
}

My config:
model {
  ssd {
    num_classes: 1
    box_coder {
      faster_rcnn_box_coder {
        y_scale: 10.0
        x_scale: 10.0
        height_scale: 5.0
        width_scale: 5.0
      }
    }
    matcher {
      argmax_matcher {
        matched_threshold: 0.5
        unmatched_threshold: 0.5
        ignore_thresholds: false
        negatives_lower_than_unmatched: true
        force_match_for_each_row: true
      }
    }
    similarity_calculator {
      iou_similarity {
      }
    }
    anchor_generator {
      ssd_anchor_generator {
        num_layers: 6
        min_scale: 0.2
        max_scale: 0.95
        aspect_ratios: 1.0
        aspect_ratios: 2.0
        aspect_ratios: 0.5
        aspect_ratios: 3.0
        aspect_ratios: 0.3333
        reduce_boxes_in_lowest_layer: true
      }
    }
    image_resizer {
      fixed_shape_resizer {
        height: 500
        width: 500
      }
    }
    box_predictor {
      convolutional_box_predictor {
        min_depth: 0
        max_depth: 0
        num_layers_before_predictor: 0
        use_dropout: false
        dropout_keep_probability: 0.5
        kernel_size: 3
        box_code_size: 4
        apply_sigmoid_to_scores: false
        conv_hyperparams {
          activation: RELU_6,
          regularizer {
            l2_regularizer {
              weight: 0.00004
            }
          }
          initializer {
            truncated_normal_initializer {
              stddev: 0.03
              mean: 0.0
            }
          }
        }
      }
    }
    feature_extractor {
      type: 'ssd_inception_v2'
      min_depth: 16
      depth_multiplier: 1.0
      conv_hyperparams {
        activation: RELU_6,
        regularizer {
          l2_regularizer {
            weight: 0.00004
          }
        }
        initializer {
          truncated_normal_initializer {
            stddev: 0.03
            mean: 0.0
          }
        }
        batch_norm {
          train: true,
          scale: true,
          center: true,
          decay: 0.9997,
          epsilon: 0.001,
        }
      }
      override_base_feature_extractor_hyperparams: true
    }
    loss {
      classification_loss {
        weighted_sigmoid {
        }
      }
      localization_loss {
        weighted_smooth_l1 {
        }
      }
      hard_example_miner {
        num_hard_examples: 3000
        iou_threshold: 0.99
        loss_type: CLASSIFICATION
        max_negatives_per_positive: 3
        min_negatives_per_image: 0
      }
      classification_weight: 1.0
      localization_weight: 1.0
    }
    normalize_loss_by_num_matches: true
    post_processing {
      batch_non_max_suppression {
        score_threshold: 1e-8
        iou_threshold: 0.6
        max_detections_per_class: 100
        max_total_detections: 100
      }
      score_converter: SIGMOID
    }
  }
}

train_config: {
  batch_size: 4
  optimizer {
    rms_prop_optimizer: {
      learning_rate: {
        exponential_decay_learning_rate {
          initial_learning_rate: 0.1
          decay_steps: 100000
          decay_factor: 0.95
        }
      }
      momentum_optimizer_value: 0.9
      decay: 0.9
      epsilon: 1.0
    }
  }
  fine_tune_checkpoint: "/content/models/research/object_detection/models/model/check/model.ckpt"
  from_detection_checkpoint: true
  # Note: The below line limits the training process to 200K steps, which we
  # empirically found to be sufficient enough to train the pets dataset. This
  # effectively bypasses the learning rate schedule (the learning rate will
  # never decay). Remove the below line to train indefinitely.
  # num_steps: 200000
  data_augmentation_options {
    random_horizontal_flip {
    }
  }
  data_augmentation_options {
    ssd_random_crop {
    }
  }
}

train_input_reader: {
  tf_record_input_reader {
    input_path: "/content/models/research/object_detection/data/train.record"
  }
  label_map_path: "/content/models/research/object_detection/data/object-detection.pbtxt"
}

eval_config: {
  num_examples: 8000
  # Note: The below line limits the evaluation process to 10 evaluations.
  # Remove the below line to evaluate indefinitely.
  max_evals: 10
}

eval_input_reader: {
  tf_record_input_reader {
    input_path: "/content/models/research/object_detection/data/test.record"
  }
  label_map_path: "/content/models/research/object_detection/data/object-detection.pbtxt"
  shuffle: false
  num_readers: 1
}

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by vingtetun** [comment]
In my experience the metrics returns -1 when no bounding boxes are found in the dataset.

For example if you have a dataset with large boxes only, you will have all the metrics with the name small and medium equals -1.
But the metric that contains large will be equal 0.0 if the algorithm fails to match the bounding boxes.

I suggest checking that your dataset is correct before anything else.

**Comment by typical-byte-world** [comment]
The problem was in the tfrecords that I generated

**Comment by madhavajay** [comment]
@typical-byte-world Hi, im having the same problem. What does your TF Record fields look like and what did you change? The weird thing is mine trains perfectly, but I get -1 for all my scores using the offline_eval_map_corloc.py tool.

**Comment by typical-byte-world** [comment]
In my case, I bad generated tfrecords. I didn't find what was a problem. You need clone tf repo with object detection api and import funcs for convert from there.
You can download my [jupyter notebook](https://github.com/typical-byte-world/data-science/blob/master/useful-notebooks/generate_tfrecords.ipynb) and try it.
And check your data and coordinates

**Comment by madhavajay** [comment]
@typical-byte-world I think I figured out my problem. Somehow I was writing the wrong class index to the built in tf record labels or something. Training and inference worked fine but benchmarking failed because the classes didnt match so I got -1 for everything. Seems to mostly work now for coco_detection_metrics but I still get errors on the pascalvoc and oid2 metrics.
